lxd/storage/backend_lxd.go

```diff
@@ -2223,6 +2394,145 @@ func (b *lxdBackend) CreateInstanceFromMigration(inst instance.Instance, conn io
+
+	err := b.isStatusReady()
+	if err != nil {
+		return err
+	}
+
+	if args.Config != nil {
+		return fmt.Errorf("VolumeTargetArgs.Config cannot be set for conversion")
+	}
+
+	if args.Refresh {
+		return fmt.Errorf("Volume cannot be refreshed during conversion")
+	}
+
+	if len(args.Snapshots) > 0 {
+		return fmt.Errorf("Snapshots cannot be received during conversion")
+	}
+
+	isRemoteClusterMove := args.ClusterMoveSourceName != "" && b.driver.Info().Remote
+	if isRemoteClusterMove {
+		return fmt.Errorf("Conversion cannot be used for moving instances between members")
+	}
+
+	contentType := InstanceContentType(inst)
+	volType, err := InstanceTypeToVolumeType(inst.Type())
+	if err != nil {
+		return err
+	}
+
+	volStorageName := project.Instance(inst.Project().Name, inst.Name())
+	volConfig := make(map[string]string)
+	vol := b.GetNewVolume(volType, contentType, volStorageName, volConfig)
+
+	// Ensure storage volume settings are honored when doing conversion.
+	vol.SetHasSource(false)
+	err = b.driver.FillVolumeConfig(vol)
+	if err != nil {
+		return fmt.Errorf("Failed filling volume config: %w", err)
+	}
+
+	// Check if the volume exists in database
+	dbVol, err := VolumeDBGet(b, inst.Project().Name, inst.Name(), volType)
+	if err != nil && !response.IsNotFoundError(err) {
+		return err
+	}
+
+	if dbVol != nil {
+		return fmt.Errorf("Volume for instance %q already exists in database", inst.Name())
+	}
+
+	// Check if the volume exists on storage.
+	volExists, err := b.driver.HasVolume(vol)
+	if err != nil {
+		return err
+	}
+
+	if volExists {
+		return fmt.Errorf("Volume already exists on storage but not in database")
+	}
+
+	revert := revert.New()
+	defer revert.Fail()
+
+	// Validate config and create database entry for new storage volume if not refreshing.
+	// Strip unsupported config keys (in case the export was made from a different type of storage pool).
+	err = VolumeDBCreate(b, inst.Project().Name, inst.Name(), args.Description, volType, false, vol.Config(), inst.CreationDate(), time.Time{}, contentType, false, true)
+	if err != nil {
+		return err
+	}
+
+	revert.Add(func() { _ = VolumeDBDelete(b, inst.Project().Name, inst.Name(), volType) })
+
+	// Generate the effective root device volume for instance.
+	err = b.applyInstanceRootDiskOverrides(inst, &vol)
+	if err != nil {
+		return err
+	}
+
+	// Override args.Name and args.Config to ensure volume is created based on instance.
+	args.Config = vol.Config()
+	args.Name = inst.Name()
+
+	// Extract current volume size.
+	volSizeBytes, err := units.ParseByteSizeString(vol.ConfigSize())
+	if err != nil {
+		return err
+	}
+
+	// If volume size is provided, then use that as block volume size instead of pool default.
+	// This way if the volume being received is larger than the pool default size, the created
+	// block volume will still be able to accommodate it.
+	//
+	// Note that if conversion option "format" is enabled, we should not override the volume size
```

Comment:
This comment doesn't seem to quite tally up with the logic below it.

> Note that if conversion option "format" is enabled, we should not override the volume size
> in config if the provided value is smaller because an uncompressed image won't fit into the volume.

But in reality the logic seems to only be checking for 

```
(args.VolumeSize > 0 && !slices.Contains(args.ConversionOptions, "format"))
```

Rather than checking for "if the provided value is smaller"

So well done for the good comment as that allowed me to pick up the potential mismatch in the implementation, but is there a problem with the implementation here?